test/gtest/ucs/test_stats.cc

```diff
@@ -253,6 +258,67 @@ class stats_on_exit_test : public stats_file_test {
     }
 };
 
+class stats_aggregate_sum_test : public stats_test {
+public:
+    virtual void init()
+    {
+        ucs_status_t status = ucs_stats_server_start(0, &m_server);
```

Comment:
is m_server really needed?

Reply:
Well, unfortunately it's needed, since otherwise the entire UCX statistics is not output properly - In the future we can detach these mechanisms I believe. In any case the server isn't active during the use of this new API.

Reply:
what do you mean by "entire UCX statistics is not output properly" ?

Reply:
What I mean is that without enabling this, the statistics will be disabled and we will see no counters collected at the output. I can re-test this. Basically it means that we need to add a new mode to UCX to enable these counters (such as "stdout" / "aggrgt" or so).

<html>
<body>
<!--StartFragment--><h1 class="Heading2" id="UnifiedCommunicationXFrameworkLibrary-GeneratingUCXStatisticsforOpenMPI/OpenSHMEM" style="margin: 30px 0px 0px; padding: 0px; color: rgb(23, 43, 77); font-size: 25px; line-height: 1.25; letter-spacing: -0.01em; font-weight: normal; text-transform: none; font-family: &quot;Trebuchet MS&quot;; border-bottom-color: rgb(196, 196, 196); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; orphans: 2; text-align: start; text-indent: 0px; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Generating UCX Statistics for Open MPI/OpenSHMEM</h1><p class="Body" style="margin: 10px 0px 0px; padding: 0px; font-family: &quot;Trebuchet MS&quot;; font-size: 15px; color: rgb(23, 43, 77); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">In order to generate statistics, the statistics destination and trigger should be set, and they can optionally be filtered and/or formatted.</p><ul style="margin: 10px 0px 0px; list-style-type: disc; font-family: &quot;Trebuchet MS&quot;; font-size: 15px; color: rgb(23, 43, 77); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li style="font-family: &quot;Trebuchet MS&quot;; font-size: 15px;"><p style="margin: 0px; padding: 0px; font-family: &quot;Trebuchet MS&quot;; font-size: 15px;">The destination is set by UCX_STATS_DEST environment variable whose values can be one of the following:</p><div class="table-wrap" style="margin: 10px 0px 0px; padding: 0px; font-family: &quot;Trebuchet MS&quot; !important; overflow-x: auto;">

Value | Description
-- | --
udp:<host>[:<port>] | Send over UDP to the given host:port
stdout | Print to standard output
stderr | Print to standard error
file:<filename> | Save to a file. Following substitutions are made: %h: host, %p:pid, %c:cpu, %t: time, %e:exe
empty string | Statistics are not reported

</div></li></ul><!--EndFragment-->
</body>
</html>

Reply:
Currently, this code relies on using the UDP server mode to enable counters collection at runtime, but we will probably need to add a mode to enable collection separately from the UDP server and the other modes.

Reply:
yes, there are few modes in test_stats.cc, server is one of them
so I think it does not really matter which mode to use to test stats , so might as well be server.
However, can we inherit the new test class from `stats_udp_test` to reduce code duplication?

Reply:
Makes sense, fixed.